src/game/SpellEffects.cpp

```diff
@@ -4594,15 +4594,17 @@ void Spell::EffectSummonPet(SpellEffectIndex eff_idx)
 
     Pet* NewSummon = new Pet;
 
+    bool skip;
+
     if (m_caster->GetTypeId() == TYPEID_PLAYER)
     {
         switch(m_caster->getClass())
         {
             case CLASS_HUNTER:
             {
-                // Everything already taken care of, we are only here because we loaded pet from db successfully
-                delete NewSummon;
-                return;
+                // Everything already taken care of, we are only here because we loaded pet from db successfully                
+                skip = NewSummon->LoadPetFromDB((Player*)m_caster, 0);;
```

Comment:
okay, here's an issue, you got two ;; ending this.
also, this check should hopefully be done in CheckCast; if the hunter has no pet he should be stopped before reaching here; he should NOT go through the summon function.